openvpn/primitives/generate.go

```diff
@@ -0,0 +1,134 @@
+package primitives
+
+import (
+	"crypto/ecdsa"
+	"crypto/x509"
+	"fmt"
+	log "github.com/cihub/seelog"
+	"os"
+	"path/filepath"
+)
+
+// SecurityPrimitives describes security primitives
+type SecurityPrimitives struct {
+	directory        string
+	CACertPath       string
+	CAKeyPath        string
+	ServerCertPath   string
+	ServerKeyPath    string
+	CRLPEMPath       string
+	TLSCryptKeyPath  string
+	caBytes          []byte
+	caPrivateKey     *ecdsa.PrivateKey
+	serverCertBytes  []byte
+	serverPrivateKey *ecdsa.PrivateKey
+}
+
+func exists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return true, err
+}
+
+func (sp *SecurityPrimitives) mkDir(dir string) {
+	if e, err := exists(dir); !e {
+		log.Debug("Creating dir (" + dir + ")")
+		os.Mkdir(dir, 0700)
+	} else {
+		log.Errorf("Error creating directory", err)
+	}
+}
+
+const (
+	genDir    = "certs"
+	caDir     = "ca"
+	serverDir = "server"
+	clientDir = "clients"
+	commonDir = "common"
+)
+
+func (sp *SecurityPrimitives) init() {
+	sp.cleanupDir()
+
+	sp.mkDir(sp.directory)
```

Comment:
Dont create new directories, rather inject&use `options.directoryRuntime`

Reply:
should I create everything in single directory, or should I create subdirectories still?

Reply:
removing all new directories..